src/modeler/AutoRest.Swagger/Validation/DisallowMsdnReferences.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class DisallowMsdnReferences : TypedRule<string>
+    {
+        private const string MsdnString = "https://msdn.microsoft.com/";
```

Comment:
you might want this a bit wider than that.
- don't only look for https
- `msdn.com` is an alias for `msdn.microsoft.com` (so links might use that)
- there may be more variants than that...